memconn_listener.go

```diff
@@ -87,24 +51,28 @@ func (l *listener) dial(
 	return localWrapper, nil
 }
 
+type errListenerClosed struct{}
```

Comment:
Could you just use `errors.New` instead of these new error types?
```go
var errListenerClosed = errors.New("closed")
```
Could also be useful to callers if exported?
```go
if err == memconn.ErrListenerClosed {
   // ignoring closed
}

Reply:
Hi @dougm,

I personally think comparing errors like that is fraught with potential for failure due to the way errors get wrapped, rereferenced , dereferenced, etc. I prefer:

```
if err != nil && err.Error() == memconn.ErrListenerClosed.Error() {
```

Regardless, I made it a struct because it follows the pattern in other Go code I've seen, takes up zero memory unless used, and can easily have special functions added to it that makes it adhere to error-related interfaces (see the `net` pkg for more info).

The other reason *not* to do `type errListenerClosed string` and then assign it to a constant string is because you're forced to type assert when you return the value from the `Error() string` function.

Reply:
Sure, I think if wrapped, etc., consider that a different error.  The 2 error types in this PR in their current form are similar to `io.EOF` as an example, and the concise use of `if err == io.EOF` can keep things simpler.  Though this kind of use case may not apply to these error types.

Reply:
Hi @dougm,

My point is that even with `io.EOF`, I don't believe it is safe to do a direct equality test since you don't necessarily know if the error will be wrapped or if, for example, some returned `errors.New` with the same value that is contained by `io.EOF`.